variables.tf

```diff
@@ -95,6 +95,56 @@ variable "managed_identities" {
   description = "Managed identities to be created for the resource."
 }
 
+variable "node_pools" {
+  type = map(object({
+    vm_size             = string
+    enable_auto_scaling = bool
+    max_count           = number
+    min_count           = number
+    node_count          = number
+    os_sku              = string
+    mode                = optional(string, "User")
+    os_disk_size_gb     = optional(number, null)
+    tags                = optional(map(string), {})
+    zones               = optional(string)
+  }))
+  default = {
```

Comment:
This is going to create 3 nodepools also in those regions that don't support AZ, correct ? Is this what we want ?

Reply:
i've added a condition to only create 1 if the zone isn't available, unfortunately i still have to keep the 3 node pool values since i can't use `count` on a list to control the generation of user node pool